src/cache.js

```diff
@@ -2,18 +2,18 @@
 
 var _cache = [];
 
-function EventHandler(token, eventName, handler, enabled) {
+function EventHandler(token, eventName, splitEventName, handler, enabled) {
   this.elementToken = token;
-  this.eventName = 'click.namespacename';
-  this.eventType = 'click';
-  this.namespace = 'namespace';
+  this.eventName = eventName;
   this.handler = handler;
   this.enabled = enabled;
+  this.eventType = splitEventName[0];
+  this.namespace = splitEventName[1];
 }
 
 module.exports = {
-  add : function(token, eventName, handler) {
-    _cache.push(new EventHandler(token, eventName, handler, true));
+  add : function(token, eventName, splitEventName, handler) {
+    _cache.push(new EventHandler(token, eventName, splitEventName, handler, true));
   },
```

Comment:
Have the add function take the descriptor and return the new EventHandler. Then the caller can access the split event name.